deepchem/feat/bio_seq_featurizer.py

```diff
@@ -0,0 +1,125 @@
+import numpy as np
+import deepchem as dc
+try:
+    import pysam
+except ImportError:
+    print("Error: Unable to import pysam. Please make sure it is installed.")
+from deepchem.feat import Featurizer
+
+class SAMFeaturizer(Featurizer):
+    """
+    This class extracts Query Name, Query Sequence, Query Length, Reference Name, 
+    Reference Start, CIGAR and Mapping Quality of the alignment in the SAM file.
+
+    Examples
+    --------
+    >>> from deepchem.data.data_loader import SAMLoader
+    >>> import deepchem as dc
+    >>> inputs = 'deepchem/data/tests/example.sam'
+    >>> featurizer = dc.feat.SAMFeaturizer()
+    >>> features = featurizer.featurize(inputs)
```

Comment:
A featurizer should be RAM to RAM and not touch disk. Let's have the featurizer accept a string buffer or some other in memory object